authorization/src/main/kotlin/com/ritense/authorization/request/DelegateUserEntityAuthorizationRequest.kt

```diff
@@ -22,9 +22,16 @@ class DelegateUserEntityAuthorizationRequest<T>(
     resourceType: Class<T>,
     action: Action<T>,
     override val user: String?,
-    entity: T?, //TODO: Determine if this really should be nullable
+    entities: List<T>
 ) : EntityAuthorizationRequest<T>(
     resourceType,
     action,
-    entity
-)
\ No newline at end of file
+    entities
+) {
+    constructor(resourceType: Class<T>, action: Action<T>, user: String?, vararg entities: T) : this(
+        resourceType,
+        action,
+        user,
+        if (entities.any { it == null }) emptyList() else entities.filterNotNull().toList()
```

Comment:
I'm not sure what the reasoning is behind the if-else, but wouldn't always filtering out the nulls have the same effect? I'm just guessing this line is here to check if any arguments have been provided?

Reply:
A 'null' or an empty list will result in a permission denied. This is because the code below should still result in an access denied:

new EntityAuthorizationRequest(
    CamundaProcessDefinition.class,
    Action.deny(),
    null
)

Although there is this use-case that is bothersome. What if the taskService returns null:

new EntityAuthorizationRequest(
    CamundaTask.class,
    VIEW,
    taskService.findTaskById(taskId)
)